### PR TITLE
refactor(gradio): decompose training interface into focused modules

### PR DESCRIPTION
**Pull request by 1larity**
﻿﻿## Summary
This PR decomposes `acestep/ui/gradio/interfaces/training.py` into focused training interface builders while preserving wiring contracts and UI behavior.

It also restores/extends i18n coverage for newly introduced training/LoKr strings across `en/ja/zh/he`, with UTF-8-safe handling to avoid mojibake and preserve emoji labels.

## What Changed
- Kept `acestep/ui/gradio/interfaces/training.py` as a thin facade/orchestrator.
- Extracted focused interface modules:
  - `training_contract_ast_utils.py`
  - `training_dataset_builder_tab.py`
  - `training_dataset_tab_scan_settings.py`
  - `training_dataset_tab_label_preview.py`
  - `training_dataset_tab_save_preprocess.py`
  - `training_lora_tab.py`
  - `training_lora_tab_dataset.py`
  - `training_lora_tab_run_export.py`
  - `training_lokr_tab.py`
  - `training_lokr_tab_dataset.py`
  - `training_lokr_tab_run_export.py`
- Added decomposition contract coverage:
  - `training_decomposition_contract_test.py`
- Updated i18n files:
  - `acestep/ui/gradio/i18n/en.json`
  - `acestep/ui/gradio/i18n/ja.json`
  - `acestep/ui/gradio/i18n/zh.json`
  - `acestep/ui/gradio/i18n/he.json`

## Behavioral Parity
- Preserved wiring key contracts expected by training event wiring.
- Restored emoji section/tab/button labels for training/LoKr UI headers and actions.
- Preserved `app.title` emoji/title text integrity (`🎛️ ... 💡`) in all touched locale files.
- Added missing LoKr/training i18n keys so decomposed UI text resolves via `t(...)` instead of hardcoded drift.

## Validation
Executed:
- `uv run python -m unittest acestep.ui.gradio.interfaces.training_decomposition_contract_test`
  - Result: `Ran 6 tests ... OK`
- `uv run python -m unittest acestep.ui.gradio.events.wiring.decomposition_contract_training_test`
  - Result: `Ran 4 tests ... OK`
- JSON parse check for touched locales (`en/ja/zh/he`)
  - Result: `i18n-json-ok`

## Scope / Out of Scope
In scope:
- training interface decomposition
- contract tests for decomposition integrity
- i18n additions required by new training/LoKr UI strings
- emoji/mojibake safety in touched locale files

Out of scope:
- non-training UI refactors
- API/runtime behavior changes
- unrelated translation rewrites outside required new keys

## Reviewer Focus
1. Facade-to-helper delegation and key parity in `training.py`.
2. Event wiring compatibility of returned training component keys.
3. UI text parity for emoji-decorated headers/buttons.
4. Locale key completeness for `en/ja/zh/he` and absence of mojibake in touched strings.

## CodeRabbit Scope Guard
Please treat findings as in-scope only when introduced by this FD + i18n patch set.
- Pre-existing issues outside touched training decomposition paths should be marked out-of-scope/non-blocking unless they indicate behavior drift introduced by this PR.

## Manual UI Validation
- Full UI test pass completed after this patch set.
- Training tabs and actions validated end-to-end with no regressions observed.
- Emoji-decorated labels/titles render correctly after i18n updates.

## Release Notes

* **New Features**
  * Added LoKr training support with dedicated tab and comprehensive configuration options for tensor decomposition, linear layers, and training parameters.
  * Introduced modularized dataset builder interface with organized sections for scanning, labeling, preprocessing, and saving.
  * Expanded internationalization support with new UI strings for Hebrew, Japanese, and Chinese languages.

* **Refactor**
  * Restructured training interface to separate concerns into focused tabs and sections for improved usability.

### CODE REVIEW

File: acestep/ui/gradio/i18n/he.json [hunk 1]
```diff
@@ -442,7 +442,37 @@
...
+    "lokr_weight_decompose_info": "Enable weight decomposition for more stable LoKr training.",
+    "lokr_learning_rate_info": "LoKr commonly uses a higher LR than LoRA. Tune per dataset.",
+    "lokr_output_dir_info": "ספרייה לשמירת משקלי LoKr המאומנים.",
+    "start_lokr_training_btn": "🚀 התחל אימון LoKr",
+    "lokr_training_loss_title": "LoKr אובדן אימון (Loss)",
+    "lokr_export_header": "ייצוא LoKr",
+    "export_lokr_btn": "📦 ייצוא LoKr",
+    "lokr_checkpoint_epoch": "Checkpoint Epoch",
+    "lokr_checkpoint_epoch_info": "Select a specific epoch checkpoint to export, or keep Latest (auto).",
+    "refresh_epochs_btn": "↻ רענון Epochs"
```
[the remark this answers is not in the document]
Comment (1larity) [reply acestep/ui/gradio/i18n/he.json:475]: Pre existing localisation/hard coded string issue. out of scope.

File: acestep/ui/gradio/i18n/zh.json [hunk 2]
```diff
@@ -426,7 +426,34 @@
...
+    "lokr_weight_decompose_info": "Enable weight decomposition for more stable LoKr training.",
+    "lokr_learning_rate_info": "LoKr commonly uses a higher LR than LoRA. Tune per dataset.",
+    "lokr_output_dir_info": "保存训练后 LoKr 权重的目录。",
+    "start_lokr_training_btn": "🚀 开始训练 LoKr",
+    "lokr_training_loss_title": "LoKr 训练损失",
+    "lokr_export_header": "导出 LoKr",
+    "export_lokr_btn": "📦 导出 LoKr",
+    "lokr_checkpoint_epoch": "Checkpoint Epoch",
+    "lokr_checkpoint_epoch_info": "Select a specific epoch checkpoint to export, or keep Latest (auto).",
+    "refresh_epochs_btn": "↻ 刷新 Epochs"
```
[the remark this answers is not in the document]
Comment (1larity) [reply acestep/ui/gradio/i18n/zh.json:456]: Pre existing localisation/hard coded string issue. out of scope.

File: acestep/ui/gradio/interfaces/training_dataset_tab_label_preview.py [hunk 3]
```diff
@@ -0,0 +1,175 @@
...
+                    info=t("training.prompt_override_info"),
+                    elem_classes=["has-info-container"],
+                )
+
+            with gr.Row():
+                edit_lyrics = gr.Textbox(
+                    label=t("training.lyrics_editable_label"),
+                    lines=6,
+                    placeholder="[Verse 1]\nLyrics here...\n\n[Chorus]\n...",
+                )
```
[the remark this answers is not in the document]
Comment (1larity) [reply acestep/ui/gradio/interfaces/training_dataset_tab_label_preview.py:95]: Pre existing localisation/hard coded string issue. out of scope.

File: acestep/ui/gradio/interfaces/training_dataset_tab_scan_settings.py [hunk 4]
```diff
@@ -0,0 +1,143 @@
...
+                    scale=3,
+                )
+                load_json_btn = gr.Button(t("training.load_btn"), variant="primary", scale=1)
+            load_json_status = gr.Textbox(
+                label=t("training.load_status"),
+                interactive=False,
+            )
+
+        with gr.Column(scale=1):
+            gr.HTML("<h4>🔍 Scan New Directory</h4>")
```
[the remark this answers is not in the document]
Comment (1larity) [reply acestep/ui/gradio/interfaces/training_dataset_tab_scan_settings.py:40]: Pre existing localisation/hard coded string issue. out of scope.

File: acestep/ui/gradio/interfaces/training_dataset_tab_scan_settings.py [hunk 5]
```diff
@@ -0,0 +1,143 @@
...
+                info="Use LM to format/structure user-provided lyrics from .txt files (coming soon)",
+                elem_classes=["has-info-container"],
+                interactive=False,
+            )
+
+            transcribe_lyrics = gr.Checkbox(
+                label="Transcribe Lyrics (LM)",
+                value=False,
+                info="Use LM to transcribe lyrics from audio (coming soon)",
+                elem_classes=["has-info-container"],
```
[the remark this answers is not in the document]
Comment (1larity) [reply acestep/ui/gradio/interfaces/training_dataset_tab_scan_settings.py:95]: Pre existing localisation/hard coded string issue. out of scope.

File: acestep/ui/gradio/interfaces/training_lora_tab_run_export.py [hunk 6]
```diff
@@ -0,0 +1,166 @@
...
+            info=t("training.output_dir_info"),
+            elem_classes=["has-info-container"],
+        )
+
+    with gr.Row():
+        resume_checkpoint_dir = gr.Textbox(
+            label="Resume Checkpoint",
+            placeholder="./lora_output/checkpoints/epoch_200",
+            info="Directory of a saved LoRA checkpoint to resume from",
+            elem_classes=["has-info-container"],
```
[the remark this answers is not in the document]
Comment (1larity) [reply acestep/ui/gradio/interfaces/training_lora_tab_run_export.py:95]: Pre existing localisation/hard coded string issue. out of scope.

File: acestep/ui/gradio/interfaces/training.py [hunk 7]
```diff
@@ -1,835 +1,68 @@
...
-            with gr.Row():
-                lokr_export_epoch = gr.Dropdown(
-                    choices=["Latest (auto)"],
-                    value="Latest (auto)",
-                    label="Checkpoint Epoch",
-                    info="Select a specific epoch checkpoint to export, or keep Latest (auto).", elem_classes=["has-info-container"],
-                )
-                refresh_lokr_export_epochs_btn = gr.Button("↻ Refresh Epochs", variant="secondary")
+    """
+    )
```
[the remark this answers is not in the document]
Comment (1larity) [reply acestep/ui/gradio/interfaces/training.py:47]: Pre existing localisation/hard coded string issue. out of scope.

File: acestep/ui/gradio/interfaces/training_dataset_tab_save_preprocess.py [hunk 8]
```diff
@@ -0,0 +1,105 @@
...
+    gr.Markdown(t("training.step5_details"))
+
+    with gr.Row():
+        preprocess_mode = gr.Dropdown(
+            label="Preprocess For",
+            choices=["LoRA", "LoKr"],
+            value="LoRA",
+            info="LoRA keeps compatibility mode; LoKr uses per-sample source-style context.",
+            elem_classes=["has-info-container"],
+        )
```
[the remark this answers is not in the document]
Comment (1larity) [reply acestep/ui/gradio/interfaces/training_dataset_tab_save_preprocess.py:70]: Pre existing localisation/hard coded string issue. out of scope.

File: acestep/ui/gradio/i18n/ja.json [hunk 9]
```diff
@@ -426,7 +426,34 @@
...
+    "lokr_weight_decompose_info": "Enable weight decomposition for more stable LoKr training.",
+    "lokr_learning_rate_info": "LoKr commonly uses a higher LR than LoRA. Tune per dataset.",
+    "lokr_output_dir_info": "トレーニング済み LoKr 重みの保存先ディレクトリ。",
+    "start_lokr_training_btn": "🚀 トレーニング開始 LoKr",
+    "lokr_training_loss_title": "LoKr トレーニング損失",
+    "lokr_export_header": "LoKr をエクスポート",
+    "export_lokr_btn": "📦 LoKr をエクスポート",
+    "lokr_checkpoint_epoch": "Checkpoint Epoch",
+    "lokr_checkpoint_epoch_info": "Select a specific epoch checkpoint to export, or keep Latest (auto).",
+    "refresh_epochs_btn": "↻ 更新 Epochs"
```
[the remark this answers is not in the document]
Comment (1larity) [reply acestep/ui/gradio/i18n/ja.json:456]: Pre existing localisation/hard coded string issue. out of scope.
